packages/venia-ui/lib/components/SavedPaymentsPage/creditCard.js

```diff
@@ -0,0 +1,85 @@
+import React, { useCallback, useMemo } from 'react';
+import { FormattedMessage } from 'react-intl';
+import { Trash2 as DeleteIcon } from 'react-feather';
+
+import { mergeClasses } from '@magento/venia-ui/lib/classify';
+
+import Icon from '../Icon';
+
+import defaultClasses from './creditCard.css';
+
+/**
+ * Enumerated list of supported credit card types from
+ *
+ * https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/Payment/view/base/web/js/model/credit-card-validation/credit-card-number-validator/credit-card-type.js
+ */
+const cardTypeMapper = {
+    AE: 'American Express',
+    AU: 'Aura',
+    DI: 'Discover',
+    DN: 'Diners',
+    ELO: 'Elo',
+    HC: 'Hipercard',
+    JCB: 'JCB',
+    MC: 'MasterCard',
+    MD: 'Maestro Domestic',
+    MI: 'Maestro International',
+    UN: 'UnionPay',
+    VI: 'Visa'
+};
```

Comment:
Is this something we can get from the backend? We would have to do a release of PWA Studio just to account for new card types added or if codes change. I don't know how often that happens, so I guess it really just depends.

Reply:
You are right about it. I have created https://jira.corp.magento.com/browse/MC-39592 on the GQL team's board to get these details from the GQL query.